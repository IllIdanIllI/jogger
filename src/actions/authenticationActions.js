import axiosInstance from './axios';
import { AUTH_URL } from '../constants/constants';
import { AUTHENTICATION, LOG_OUT, RE_AUTHENTICATION } from '../constants/types';

export const authenticate = () => (dispatch) => {
    const body = JSON.stringify({ uuid: 'hello' });
    axiosInstance.post(`${AUTH_URL}/uuidLogin`, body).then((response) =>
        dispatch({
            type: AUTHENTICATION,
            payload: response.data.response.access_token,
        })
    );
};

export const reAuthenticate = (token) => (dispatch) => {
    dispatch({
        type: RE_AUTHENTICATION,
        payload: token,
    });
};

export const logOut = () => (dispatch) => {
    dispatch({
        type: LOG_OUT,
    });
};
