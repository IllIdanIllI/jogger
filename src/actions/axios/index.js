import axios from 'axios';

export const COMMON_URL = `https://jogtracker.herokuapp.com/api/v1`;
const axiosInstance = 
    axios.create({
        baseURL: COMMON_URL,
        headers: { 'Content-Type': 'application/json' },
    });
export default axiosInstance;
