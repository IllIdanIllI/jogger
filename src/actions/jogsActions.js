import axiosInstance, { authorizationHeader } from './axios';
import { DATA_URL, TOKEN } from '../constants/constants';
import { RECEIVE_JOGS, TOGGLE_FILTER } from '../constants/types';
import { dateFromTimestampToInput } from '../service/dataService';

export const receiveJogs = () => (dispatch) => {
    const token = localStorage.getItem(TOKEN);
    axiosInstance
        .get(`${DATA_URL}/sync`, {
            headers: { Authorization: authorizationHeader(token) },
        })
        .then((response) =>
            dispatch({
                type: RECEIVE_JOGS,
                payload: response.data.response,
            })
        );
};

export const setJogs = (dateTo, dateFrom) => (dispatch) => {
    const token = localStorage.getItem(TOKEN);
    axiosInstance
        .get(`${DATA_URL}/sync`, {
            headers: { Authorization: authorizationHeader(token) },
        })
        .then((response) => {
            const payload = response.data.response;
            payload.jogs = payload.jogs.filter(
                (jog) =>
                    dateFromTimestampToInput(jog.date) <= dateTo &&
                    dateFromTimestampToInput(jog.date) >= dateFrom
            );
            dispatch({
                type: RECEIVE_JOGS,
                payload: response.data.response,
            });
        });
};

export const toggleFilter = (state) => (dispatch) => {
    dispatch({
        type: TOGGLE_FILTER,
        payload: !state,
    });
};

export const addJog = (jog) => async (dispatch) => {
    const token = localStorage.getItem(TOKEN);
    const body = JSON.stringify(jog);
    await axiosInstance.post(`${DATA_URL}/jog`, body, {
        headers: { Authorization: authorizationHeader(token) },
    });
};

export const updateJog = (jog) => async (dispatch) => {
    const token = localStorage.getItem(TOKEN);
    const body = JSON.stringify(jog);
    await axiosInstance.put(`${DATA_URL}/jog`, body, {
        headers: { Authorization: authorizationHeader(token) },
    });
};
