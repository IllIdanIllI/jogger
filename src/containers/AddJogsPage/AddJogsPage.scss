@import '../../style/colors';

$buttonHeight: 60px;

.add-jogs-container {
    position: relative;
    top: calc(116px + 10vh);
    justify-items: center;

    &__window {
        margin: auto;
        width: 503px;
        height: 379px;
        border-radius: 44px;
        background-color: $apple-green;

        &-cross {
            cursor: pointer;
            position: relative;
            float: right;
            top: 24px;
            right: 27px;
        }

        &-form {
            display: grid;
            top: 90px;
            justify-items: center;
            &__inputs {
                display: grid;
                grid-template-columns: min-content 236px;
                grid-column-gap: 16px;
                grid-row-gap: 25px;
                align-items: center;
                input {
                    padding-left: 8px;
                    width: 236px;
                    font-family: SFUIText;
                    height: 31px;
                    border-radius: 7px;
                    border: solid 1px #979797;
                    background-color: $white-two;
                }
                label {
                    font-size: 16px;
                }
            }
            &__button {
                margin-top: 46px;
                cursor: pointer;
                background-color: transparent;
                width: 320px;
                height: 42px;
                border-radius: 25.2px;
                border: solid 2px $white-two;
                span {
                    color: $white-two;
                    font-size: 12.6px;
                    font-weight: bold;
                    text-align: center;
                }
            }
        }
    }
}

@media screen and (max-width: 800px) {
    .add-jogs-container {
        top: calc(116px + 10vh);
        // justify-items: center;

        &__window {
            margin: auto;
            width: 90vw;
            height: 60vh;
            border-radius: 44px;
            background-color: $apple-green;

            &-form {
                margin: 0;
                top: 10%;
                height: 100%;
                width: 100%;
                grid-template-rows: min-content min-content;
                grid-row-gap: 40px;

                &__inputs {
                    display: grid;
                    grid-template-rows: repeat(6, min-content);
                    grid-template-columns: unset;
                    width: 90%;
                    grid-row-gap: 15px;
                    justify-items: center;
                    align-items: center;

                    input {
                        float: left;
                        padding-left: 8px;
                        left: 5%;
                        width: 100%;
                        font-family: SFUIText;
                        height: 31px;
                        border-radius: 7px;
                        border: solid 1px #979797;
                        background-color: $white-two;
                    }
                    label {
                        justify-self: self-start;
                        height: 14px;
                        font-size: 14px;
                    }
                }
                &__button {
                    margin-top: 0;
                    span {
                        color: $white-two;
                        font-size: 12.6px;
                        font-weight: bold;
                        text-align: center;
                    }
                }
            }
        }
    }
}
