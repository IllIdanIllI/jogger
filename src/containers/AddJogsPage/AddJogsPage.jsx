import React, { useState, useMemo } from 'react';
import { useSelector } from 'react-redux';
import { Link, useLocation, useHistory } from 'react-router-dom';
import './AddJogsPage.scss';
import { ReactComponent as Cross } from '../../images/cancel/cancel.svg';
import { JOGS_URL, ADD_URL } from '../../constants/constants';
import { useActions } from '../../custom/customHooks';
import { addJogs } from '../../actions/jogsActions';
import { dateToInput } from '../../service/dataService';

const AddJogsPage = () => {
    const defineDate = useMemo(() => dateToInput(), []);

    const location = useLocation();
    const history = useHistory();

    const isUpdate = useMemo(
        () => (location.pathname === JOGS_URL + ADD_URL ? false : true),
        [location.pathname]
    );

    const formData = useSelector((state) => state.jogStore.jogs[0]);

    const [onAddJogs] = useActions([addJogs]);

    const [distance, setDistance] = useState(isUpdate ? formData.distance : 0);
    const [time, setTime] = useState(isUpdate ? formData.time : 0);
    const [date, setDate] = useState(isUpdate ? formData.date : defineDate);

    const onUpdateJog = () => {};

    const addJogHandler = () => {
        onAddJogs({ distance, time, date });
        setTimeout(() => history.push(JOGS_URL));
    };

    return (
        <div className="add-jogs-container">
            <div className="add-jogs-container__window">
                <Link to={JOGS_URL}>
                    <Cross className="add-jogs-container__window-cross" />
                </Link>

                <div className="add-jogs-container__window-form">
                    <div className="add-jogs-container__window-form__inputs">
                        <label htmlFor="Distance">Distance</label>
                        <input
                            type="number"
                            onChange={(e) => setDistance(e.target.value)}
                            value={distance}
                            id="Distance"
                        />
                        <label htmlFor="Time">Time</label>
                        <input
                            type="number"
                            onChange={(e) => setTime(e.target.value)}
                            value={time}
                            id="Time"
                        />
                        <label htmlFor="Date">Date</label>
                        <input
                            type="date"
                            onChange={(e) => setDate(e.target.value)}
                            value={date}
                            id="Date"
                        />
                    </div>
                    {isUpdate ? (
                        <button
                            onClick={() => onUpdateJog()}
                            className="add-jogs-container__window-form__button"
                        >
                            <span>Update</span>
                        </button>
                    ) : (
                        <button
                            onClick={() => addJogHandler()}
                            className="add-jogs-container__window-form__button"
                        >
                            <span>Save</span>
                        </button>
                    )}
                </div>
            </div>
        </div>
    );
};

export default AddJogsPage;
