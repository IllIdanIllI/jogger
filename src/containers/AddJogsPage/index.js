import AddJogsPage from './AddJogsPage';

export default AddJogsPage;
