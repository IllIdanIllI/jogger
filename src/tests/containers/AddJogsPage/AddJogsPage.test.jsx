import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import AddJogsPage from '../../../containers/AddJogsPage';
import { updateJog } from '../../../actions/jogsActions';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
const mockStore = configureMockStore([thunk]);

describe('<AddJogsPage /> unit test', () => {
    const initialState = {
        authentication: { isAuthenticated: true },
        jogStore: { isFilterActive: false },
    };
    // const mockStore = configureStore();
    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(
            <Provider store={store}>
                <Router>
                    <AddJogsPage />
                </Router>
            </Provider>
        );
    });
    it('Simulate update', async () => {
        const newJog = {
            date: 3,
            time: 3,
            distance: 3,
            jog_id: 1,
            user_id: 1,
        };
        try {
            await store.dispatch(updateJog(newJog));
        } catch (e) {
            expect(e.response.status).toEqual(401);
        }
    });

    it('Match snap', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
