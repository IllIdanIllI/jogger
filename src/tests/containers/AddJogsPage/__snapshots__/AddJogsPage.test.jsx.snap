// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddJogsPage /> unit test Match snap 1`] = `ReactWrapper {}`;
